src/Libraries/Microsoft.Extensions.AI.Abstractions/Tools/HostedMcpServerTool.cs

```diff
@@ -134,4 +161,18 @@ private static string ValidateUrl(Uri serverUrl)
     /// </para>
     /// </remarks>
     public HostedMcpServerToolApprovalMode? ApprovalMode { get; set; }
+
+    /// <summary>
+    /// Gets or sets a mutable dictionary of HTTP headers to include when calling the remote MCP server.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The underlying provider is not guaranteed to support or honor the headers.
+    /// </para>
+    /// </remarks>
+    public IDictionary<string, string> Headers
+    {
+        get => field ??= new Dictionary<string, string>();
```

Comment:
Probably better to use a named private backing field. Then Authorization Token can check that field instead of the property and avoid forcing the headers dictionary into existence for a common case of there not being sent headers